### PR TITLE
(QENG-1841) updated AIO pathing for linux & windows

### PR DESCRIPTION
**Pull request by kevpl**

### CODE REVIEW

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
just known failures, unrelated to change

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

File: lib/beaker/host/windows.rb [hunk 1]
```diff
@@ -50,12 +50,17 @@ def self.foss_defaults
...
         'hierabindir'       => '/opt/puppet-git-repos/hiera/bin',
-        'pathseparator'     => ':',
+        'pathseparator'     => ';',
       })
     end
 
     def self.aio_defaults
-      self.foss_defaults
+      h = self.foss_defaults
+      h['puppetvardir']   = '`cygpath -smF 35`/PuppetLabs/puppet/cache'
```
Comment (Iristyle) [inline lib/beaker/host/windows.rb]: OK - a couple quick notes.  Essentially, 2 main categories of things are incorrect:

* The `codedir` should be updated inline with the spec - https://github.com/puppetlabs/puppet-specifications/blob/master/file_paths.md#puppet-agent-windows

* The `foss_defaults` are using Git style locations for hiera config values, even though AIO installs from the MSI.  I believe these can be omitted.

```ruby
h['puppetcodedir']     = '`cygpath -smF 35`/PuppetLabs/code'

['hieralibdir', 'hierapuppetlibdir', 'hierabindir'].each { |k| h.delete(k) }
```

Since this would now start trashing things from an existing hash, I think it's probably better to just start with a fresh hash, and explicitly define all the values like is done for `pe_defaults` and `foss_defaults`

File: lib/beaker/host/unix.rb [hunk 2]
```diff
@@ -64,16 +64,16 @@ def self.aio_defaults
       h = self.foss_defaults
       h['puppetserver-confdir'] = '/etc/puppetlabs/puppetserver/conf.d'
       h['puppetservice']  = 'puppetserver'
-      h['puppetbindir']   = '/opt/puppetlabs/agent/bin'
+      h['puppetbindir']   = '/opt/puppetlabs/puppet/bin'
```
Comment (Iristyle) [inline lib/beaker/host/unix.rb]: @joshcooper / @haus - better to use `/opt/puppetlabs/puppet/bin` or `/opt/puppetlabs/bin` which has symlinks?

File: lib/beaker/host/unix.rb [hunk 3]
```diff
@@ -64,16 +64,16 @@ def self.aio_defaults
...
       h['puppetbin']      = "#{h['puppetbindir']}/puppet"
-      h['puppetpath']     = '/etc/puppetlabs/agent'
-      h['puppetconfdir']  = "#{h['puppetpath']}/config"
-      h['puppetcodedir']  = "#{h['puppetpath']}/code"
-      h['puppetvardir']   = '/opt/puppetlabs/agent/cache'
+      h['puppetpath']     = '/etc/puppetlabs/puppet'
+      h['puppetconfdir']  = "#{h['puppetpath']}"
```
Comment (Iristyle) [inline lib/beaker/host/unix.rb]: String interpolation not necessary here:

```ruby
h['puppetconfdir'] = h['puppetpath']
```

File: lib/beaker/host/unix.rb [hunk 4]
```diff
@@ -64,16 +64,16 @@ def self.aio_defaults
...
+      h['puppetcodedir']  = '/etc/puppetlabs/code'
       h['distmoduledir']  = "#{h['puppetcodedir']}/modules"
-      h['sitemoduledir']  = '/opt/puppetlabs/agent/modules'
       h['hieraconf']      = "#{h['puppetcodedir']}/hiera.yaml"
       h['hieradatadir']   = "#{h['puppetcodedir']}/hieradata"
+      h['puppetvardir']   = '/opt/puppetlabs/puppet/cache'
```
Comment (Iristyle) [inline lib/beaker/host/unix.rb]: Nit - re-ordering settings throws off diff.

File: lib/beaker/host/unix.rb (hunk 2 above)
Comment (joshcooper) [inline lib/beaker/host/unix.rb]: @kevpl we had a discussion about this, and we want beaker to be using the "public" AIO bin directory which is `/opt/puppetlabs/bin`.

There are some puppet acceptance tests, e.g. https://github.com/puppetlabs/puppet/blob/master/acceptance/tests/environment/use_enc_environment.rb#L18, that assume `puppetbindir` points to the "private" bin directory, which contains ruby. We are going to update the tests on our side, but don't let that block beaker.

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

File: lib/beaker/host/windows.rb [hunk 5]
```diff
@@ -50,12 +50,29 @@ def self.foss_defaults
...
-        'pathseparator'     => ':',
+        'pathseparator'     => ';',
       })
     end
 
     def self.aio_defaults
-      self.foss_defaults
+      h = Beaker::Options::OptionsHash.new
+      h.merge({
+        'user'              => 'Administrator',
```
Comment (Iristyle) [inline lib/beaker/host/windows.rb]: `user` and `group` changes seem unrelated to QENG-1841 - pathing changes.  Not sure I understand how these are related?

File: lib/beaker/host/windows.rb [hunk 6]
```diff
@@ -29,7 +29,7 @@ def self.pe_defaults
         'sitemoduledir' => 'C:/usr/share/puppet/modules',
         #let's just add both potential bin dirs to the path
         'puppetbindir'  => '/cygdrive/c/Program Files (x86)/Puppet Labs/Puppet Enterprise/bin:/cygdrive/c/Program Files/Puppet Labs/Puppet Enterprise/bin',
-        'pathseparator' => ':',
+        'pathseparator' => ';',
```
Comment (Iristyle) [inline lib/beaker/host/windows.rb]: You might also want to add a note about which commit introduced this errant change (and why it's being undone in this commit)

File: lib/beaker/host/unix.rb [hunk 7]
```diff
@@ -61,20 +61,24 @@ def self.foss_defaults
...
-      h['sitemoduledir']  = '/opt/puppetlabs/agent/modules'
-      h['hieraconf']      = "#{h['puppetcodedir']}/hiera.yaml"
-      h['hieradatadir']   = "#{h['puppetcodedir']}/hieradata"
-      h
+      h = Beaker::Options::OptionsHash.new
+      h.merge({
+        'user'                  => 'root',
+        'group'                 => 'puppet',
+        'puppetserver-confdir'  => '/etc/puppetlabs/puppetserver/conf.d',
+        'puppetservice'         => 'puppetserver',
```
Comment (Iristyle) [inline lib/beaker/host/unix.rb]: Same note about what does this have to do with QENG-1841 / pathing changes?  At the very least, it should probably be in it's own commit / there should be something in the commit message indicating why these changes are being mixed in.

Comment (kevpl) [inline lib/beaker/host/unix.rb]: they're not actually mix-ins, they're things that we were overriding that we're not anymore.  They're actually things that aren't changed at all.

**Comment by kevpl** [comment]
all known failures, and unrelated to merge.  The `codedir` failure also showed up in this unrelated PR: https://github.com/puppetlabs/beaker/pull/726
